include/deal.II/matrix_free/operators.h

```diff
@@ -160,6 +160,12 @@ namespace MatrixFreeOperators
     virtual void compute_diagonal () = 0;
 
     /**
+     * Get read access to the MatrixFree object stored with this operator.
+     */
+    const std_cxx11::shared_ptr<const MatrixFree<dim,Number> > &
```

Comment:
@davydden In general the reference has the advantage of not incrementing (and decrementing when the returned object goes out of scope) the reference count upon call. On the other hand, this is a big object and almost any operation on it is going to be much more expensive, so we might as well stick with returning a copy, so we do not need to worry about the validity of the pointer. If you agree, I will change.